Public/Src/Tools/Xldb.Analyzer/XldbAnalyzer.cs

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Xldb.Proto;
+
+namespace Xldb.Analyzer
+{
+    /// <summary>
+    /// Xldb Analyzers that use the rocksDb instanceand the Xldb API to analyze BXL logs
+    /// </summary>
+    public class Program
+    {
+
+        private const string s_eventStatsAnalyzer = "eventstats";
+        private const string s_dumpPipAnalyzer = "dumppip";
+
+        private Dictionary<string, string> m_commandLineOptions = new Dictionary<string, string>();
+
+        public static int Main(string[] args)
+        {
+            var p = new Program();
+
+            if (!p.ProcessArguments(args))
+            {
+                Console.WriteLine("Invalid arguments passed in, exiting analyzer ...");
+                return 1;
+            }

```

Comment:
We probably don't need to be fancy with multiple mode support. I thought our goal here was to do a demonstration HelloWorld for consuming xlgdb, rather than providing scaffolding for replicating the existing analyzer model on top of a different backend. Is that still the idea?

Reply:
I think this is a demonstration with just 2 "modes" that serve as 2 demos for people to follow/use. It is up to them if they want to add their new analyzers/queries here, or keep it completely separate from our code base. We do not enforce anything or suggest it either. 

---
In reply to: [316299007](https://github.com/microsoft/BuildXL/pull/775#discussion_r316299007) [](ancestors = 316299007)